nodeup/pkg/model/kubelet.go

```diff
@@ -597,54 +594,15 @@ func (b *KubeletBuilder) buildMasterKubeletKubeconfig() (*nodetasks.File, error)
 		return nil, fmt.Errorf("error getting NodeName: %v", err)
 	}
 
-	caCert, err := b.KeyStore.FindCert(fi.CertificateId_CA)
-	if err != nil {
-		return nil, fmt.Errorf("error fetching CA certificate from keystore: %v", err)
-	}
-	if caCert == nil {
-		return nil, fmt.Errorf("unable to find CA certificate %q in keystore", fi.CertificateId_CA)
-	}
-
-	caKey, err := b.KeyStore.FindPrivateKey(fi.CertificateId_CA)
-	if err != nil {
-		return nil, fmt.Errorf("error fetching CA certificate from keystore: %v", err)
-	}
-	if caKey == nil {
-		return nil, fmt.Errorf("unable to find CA key %q in keystore", fi.CertificateId_CA)
-	}
-
-	privateKey, err := pki.GeneratePrivateKey()
-	if err != nil {
-		return nil, err
-	}
-
-	template := &x509.Certificate{
-		BasicConstraintsValid: true,
-		IsCA:                  false,
+	req := &pki.IssueCertRequest{
+		Name:         "master-kubelet",
+		Signer:       fi.CertificateId_CA,
+		Type:         "client",
+		Subject:      fmt.Sprintf("cn=system:node:%s,o=%s", nodeName, rbac.NodesGroup),
```

Comment:
This sort of thing is making me reconsider specifying the Subject as a string. One hopes there's no node name or nodeGroup with a comma in it. I'm considering changing it so that Subject has to be specified as a `pkix.Name` by all callers.